libraries/Microsoft.Bot.Connector/Authentication/JwtTokenValidation.cs

```diff
@@ -186,6 +164,22 @@ public static string GetAppIdFromClaims(IEnumerable<Claim> claims)
             return appId;
         }
 
+        /// <summary>
+        /// Validates the identity claims against the <see cref="ClaimsValidator"/> in <see cref="AuthenticationConfiguration"/> if present. 
+        /// </summary>
+        /// <param name="authConfig">An <see cref="AuthenticationConfiguration"/> instance.</param>
+        /// <param name="claims">The list of claims to validate.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        /// <exception cref="UnauthorizedAccessException">If the validation returns false.</exception>
+        internal static async Task ValidateClaimsAsync(AuthenticationConfiguration authConfig, IEnumerable<Claim> claims)
+        {
+            if (authConfig.ClaimsValidator != null && !await authConfig.ClaimsValidator.ValidateClaimsAsync(new List<Claim>(claims)).ConfigureAwait(false))
```

Comment:
This makes me think that we'll never know what is failing in the claims validator. Would be nice to have the claims validator to have a way of surfacing error information.

Reply:
I could change this to throw an exception rather than a bool, but wanted to make it consistent with other IsXyz methods we have in JwtValidation. We can discuss options offline.

Reply:
Updated code so it is responsibility of the dev that implements the validator to throw an exception when the validation fails.